src/game_api/layer.cpp

```diff
@@ -173,3 +173,32 @@ Entity* Layer::spawn_apep(float x, float y, bool right)
 
     return apep_head;
 }
+
+void Layer::move_grid_entity(Entity* ent, float x, float y, Layer* dest_layer)
+{
+    move_grid_entity(ent, static_cast<uint32_t>(std::round(x)), static_cast<uint32_t>(std::round(y)), dest_layer);
+}
+
+void Layer::move_grid_entity(Entity* ent, uint32_t x, uint32_t y, Layer* dest_layer)
+{
+    if (ent)
+    {
+        if (ent->type->search_flags & 0x100) // MASK.FLOOR

```

Comment:
What's up here? Why move only floors?

Reply:
Only floors are grid entities as far i know

Reply:
But users can spawn anything as a grid entity via the API, right?

Reply:
Nope, only get_grid_entity and move_grid_entity access `layer->grid_entities`, and it's not exposed to lua, spawn_grid_entity uses the standard spawn function

Reply:
But doesn't that standard spawn function, when called with the right params, place the spawned entity into the grid? Not in our code but in game code.

Reply:
Don't think so? what would be the reason for that? spawning the floor with standard parameters adds it to the grid
also have you tested if that even does anything when adding non floor entity to the grid?

Reply:
No, that's why I'm asking you. You're currently working with this so it's only natural that you would answer questions about it  😉